QueryGenie/Realm/ResultsWrapper.swift

```diff
@@ -0,0 +1,323 @@
+//
+//  ResultsWrapper.swift
+//  QueryGenie
+//
+//  Created by Dominic on 11/9/17.
+//
+
+import Foundation
+
+import RealmSwift
+
+/**
+Wrapper class for RealmSwift's `Results<Element>` class that sets the `Element` to `Object`.
+ 
+ - Warning: This class passes through calls to all the methods and properties on `Results` except calls to
+````
+ static func bridging(from objectiveCValue: Any, with metadata: Any?) -> ResultsWrapper
+````
+and
+````
+ var bridged: (objectiveCValue: Any, metadata: Any?)
+````
+ - Note: `Results<Element>` allows `AnyObject`, setting the `Element` to `Object`
+         allows QueryGenie extensions to work with `Results`.
+*/
+public final class ResultsWrapper<T> where T: Object {
```

Comment:
While this works, it does make the API less elegant. Now you have to constantly wrap your `Results` in this wrapper. Is there a way we could only make `Results` conform to our protocol when `Results.T: Object`?

It's been a while since I've gotten my head into this code base, but I feel like there should be a more elegant solution than this.

Reply:
I agree that there could be a more elegant solution. I will spend some more time thinking about this tomorrow. I would love to get away from the wrapper.

Reply:
If you want to take a crack at this, here are the problems we ran up against while developing the wrapper: (1) `Results` cannot be sub-classed. (2) Swift does not yet support conditional protocol conformance: https://github.com/apple/swift/blob/master/docs/GenericsManifesto.md